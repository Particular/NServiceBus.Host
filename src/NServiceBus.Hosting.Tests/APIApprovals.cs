using System.Runtime.CompilerServices;
using NServiceBus.Hosting.Tests;
using NServiceBus.Hosting.Windows;
using NUnit.Framework;
using PublicApiGenerator;

[TestFixture]
public class APIApprovals
{
    [Test]
    [MethodImpl(MethodImplOptions.NoInlining)]
    public void Approve()
    {
        var publicApi = ApiGenerator.GeneratePublicApi(typeof(Program).Assembly);
        TestApprover.Verify(publicApi);
    }
}