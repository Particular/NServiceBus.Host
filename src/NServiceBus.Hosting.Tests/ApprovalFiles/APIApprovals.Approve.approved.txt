[assembly: System.CLSCompliant(false)]
[assembly: System.Runtime.CompilerServices.InternalsVisibleTo(@"NServiceBus.Hosting.Tests, PublicKey=0024000004800000940000000602000000240000525341310004000001000100dde965e6172e019ac82c2639ffe494dd2e7dd16347c34762a05732b492e110f2e4e2e1b5ef2d85c848ccfb671ee20a47c8d1376276708dc30a90ff1121b647ba3b7259a6bc383b2034938ef0e275b58b920375ac605076178123693c6c4f1331661a62eba28c249386855637780e3ff5f23a6d854700eaa6803ef48907513b92")]
[assembly: System.Runtime.InteropServices.ComVisible(false)]
namespace NServiceBus
{
    [System.Obsolete("The windows host has been deprecated. It is recommended to switch to self-hosting" +
        " or the generic host instead. See the upgrade guide and docs for alternatives. W" +
        "ill be removed in version 10.0.0.", true)]
    public static class EndpointConfigurationExtensions
    {
        [System.Obsolete("The windows host has been deprecated. It is recommended to switch to self-hosting" +
            " or the generic host instead. See the upgrade guide and docs for alternatives. T" +
            "he member currently throws a NotImplementedException. Will be removed in version" +
            " 10.0.0.", true)]
        public static void DefineEndpointName(this NServiceBus.EndpointConfiguration configuration, string endpointName) { }
    }
    [System.AttributeUsage(System.AttributeTargets.Class | System.AttributeTargets.All)]
    [System.Obsolete("The windows host has been deprecated. It is recommended to switch to self-hosting" +
        " or the generic host instead. See the upgrade guide and docs for alternatives. W" +
        "ill be removed in version 10.0.0.", true)]
    public sealed class EndpointNameAttribute : System.Attribute
    {
        public EndpointNameAttribute(string name) { }
        public string Name { get; }
    }
    [System.Obsolete("The windows host has been deprecated. It is recommended to switch to self-hosting" +
        " or the generic host instead. See the upgrade guide and docs for alternatives. W" +
        "ill be removed in version 10.0.0.", true)]
    public static class EndpointStartableAndStoppableExtensions
    {
        [System.Obsolete("The windows host has been deprecated. It is recommended to switch to self-hosting" +
            " or the generic host instead. See the upgrade guide and docs for alternatives. T" +
            "he member currently throws a NotImplementedException. Will be removed in version" +
            " 10.0.0.", true)]
        public static void RunWhenEndpointStartsAndStops(this NServiceBus.EndpointConfiguration configuration, NServiceBus.IWantToRunWhenEndpointStartsAndStops startableAndStoppable) { }
    }
    [System.Obsolete("The windows host has been deprecated. It is recommended to switch to self-hosting" +
        " or the generic host instead. See the upgrade guide and docs for alternatives. W" +
        "ill be removed in version 10.0.0.", true)]
    public interface IConfigureThisEndpoint
    {
        [System.Obsolete("The windows host has been deprecated. It is recommended to switch to self-hosting" +
            " or the generic host instead. See the upgrade guide and docs for alternatives. W" +
            "ill be removed in version 10.0.0.", true)]
        void Customize(NServiceBus.EndpointConfiguration configuration);
    }
    [System.Obsolete("The windows host has been deprecated. It is recommended to switch to self-hosting" +
        " or the generic host instead. See the upgrade guide and docs for alternatives. W" +
        "ill be removed in version 10.0.0.", true)]
    public interface IProfile { }
    [System.Obsolete("The windows host has been deprecated. It is recommended to switch to self-hosting" +
        " or the generic host instead. See the upgrade guide and docs for alternatives. W" +
        "ill be removed in version 10.0.0.", true)]
    public interface IWantTheListOfActiveProfiles
    {
        [System.Obsolete("The windows host has been deprecated. It is recommended to switch to self-hosting" +
            " or the generic host instead. See the upgrade guide and docs for alternatives. W" +
            "ill be removed in version 10.0.0.", true)]
        System.Collections.Generic.IEnumerable<System.Type> ActiveProfiles { get; set; }
    }
    [System.Obsolete("The windows host has been deprecated. It is recommended to switch to self-hosting" +
        " or the generic host instead. See the upgrade guide and docs for alternatives. W" +
        "ill be removed in version 10.0.0.", true)]
    public interface IWantToRunWhenEndpointStartsAndStops
    {
        [System.Obsolete("The windows host has been deprecated. It is recommended to switch to self-hosting" +
            " or the generic host instead. See the upgrade guide and docs for alternatives. W" +
            "ill be removed in version 10.0.0.", true)]
        System.Threading.Tasks.Task Start(NServiceBus.IMessageSession session);
        [System.Obsolete("The windows host has been deprecated. It is recommended to switch to self-hosting" +
            " or the generic host instead. See the upgrade guide and docs for alternatives. W" +
            "ill be removed in version 10.0.0.", true)]
        System.Threading.Tasks.Task Stop(NServiceBus.IMessageSession session);
    }
    [System.Obsolete("The windows host has been deprecated. It is recommended to switch to self-hosting" +
        " or the generic host instead. See the upgrade guide and docs for alternatives. W" +
        "ill be removed in version 10.0.0.", true)]
    public interface Integration : NServiceBus.IProfile { }
    [System.Obsolete("The windows host has been deprecated. It is recommended to switch to self-hosting" +
        " or the generic host instead. See the upgrade guide and docs for alternatives. W" +
        "ill be removed in version 10.0.0.", true)]
    public interface Lite : NServiceBus.IProfile { }
    [System.Obsolete("The windows host has been deprecated. It is recommended to switch to self-hosting" +
        " or the generic host instead. See the upgrade guide and docs for alternatives. W" +
        "ill be removed in version 10.0.0.", true)]
    public interface Production : NServiceBus.IProfile { }
}
namespace NServiceBus.Hosting.Profiles
{
    [System.Obsolete("The windows host has been deprecated. It is recommended to switch to self-hosting" +
        " or the generic host instead. See the upgrade guide and docs for alternatives. W" +
        "ill be removed in version 10.0.0.", true)]
    public interface IHandleAnyProfile : NServiceBus.Hosting.Profiles.IHandleProfile, NServiceBus.Hosting.Profiles.IHandleProfile<NServiceBus.IProfile>, NServiceBus.IWantTheListOfActiveProfiles { }
    [System.Obsolete("The windows host has been deprecated. It is recommended to switch to self-hosting" +
        " or the generic host instead. See the upgrade guide and docs for alternatives. W" +
        "ill be removed in version 10.0.0.", true)]
    public interface IHandleProfile
    {
        [System.Obsolete("The windows host has been deprecated. It is recommended to switch to self-hosting" +
            " or the generic host instead. See the upgrade guide and docs for alternatives. W" +
            "ill be removed in version 10.0.0.", true)]
        void ProfileActivated(NServiceBus.EndpointConfiguration config);
    }
    [System.Obsolete("The windows host has been deprecated. It is recommended to switch to self-hosting" +
        " or the generic host instead. See the upgrade guide and docs for alternatives. W" +
        "ill be removed in version 10.0.0.", true)]
    public interface IHandleProfile<T> : NServiceBus.Hosting.Profiles.IHandleProfile
        where T : NServiceBus.IProfile { }
}
namespace NServiceBus.Hosting.Windows
{
    [System.Obsolete("The windows host has been deprecated. It is recommended to switch to self-hosting" +
        " or the generic host instead. See the upgrade guide and docs for alternatives. W" +
        "ill be removed in version 10.0.0.", true)]
    public class InstallWindowsHost : System.MarshalByRefObject
    {
        [System.Obsolete("The windows host has been deprecated. It is recommended to switch to self-hosting" +
            " or the generic host instead. See the upgrade guide and docs for alternatives. T" +
            "he member currently throws a NotImplementedException. Will be removed in version" +
            " 10.0.0.", true)]
        public InstallWindowsHost(System.Type endpointType, string[] args, string endpointName, System.Collections.Generic.IEnumerable<string> scannableAssembliesFullName) { }
        [System.Obsolete("The windows host has been deprecated. It is recommended to switch to self-hosting" +
            " or the generic host instead. See the upgrade guide and docs for alternatives. T" +
            "he member currently throws a NotImplementedException. Will be removed in version" +
            " 10.0.0.", true)]
        public void Install(string username) { }
    }
    [System.Obsolete("The windows host has been deprecated. It is recommended to switch to self-hosting" +
        " or the generic host instead. See the upgrade guide and docs for alternatives. W" +
        "ill be removed in version 10.0.0.", true)]
    public class Program
    {
        public Program() { }
    }
    [System.Obsolete("The windows host has been deprecated. It is recommended to switch to self-hosting" +
        " or the generic host instead. See the upgrade guide and docs for alternatives. W" +
        "ill be removed in version 10.0.0.", true)]
    public class WindowsHost : System.MarshalByRefObject
    {
        [System.Obsolete("The windows host has been deprecated. It is recommended to switch to self-hosting" +
            " or the generic host instead. See the upgrade guide and docs for alternatives. T" +
            "he member currently throws a NotImplementedException. Will be removed in version" +
            " 10.0.0.", true)]
        public WindowsHost(System.Type endpointType, string[] args, string endpointName, System.Collections.Generic.IEnumerable<string> scannableAssembliesFullName) { }
        [System.Obsolete("The windows host has been deprecated. It is recommended to switch to self-hosting" +
            " or the generic host instead. See the upgrade guide and docs for alternatives. T" +
            "he member currently throws a NotImplementedException. Will be removed in version" +
            " 10.0.0.", true)]
        public void Start() { }
        [System.Obsolete("The windows host has been deprecated. It is recommended to switch to self-hosting" +
            " or the generic host instead. See the upgrade guide and docs for alternatives. T" +
            "he member currently throws a NotImplementedException. Will be removed in version" +
            " 10.0.0.", true)]
        public void Stop() { }
    }
}